packages/next/src/client/set-attributes-from-props.ts

```diff
@@ -0,0 +1,59 @@
+const DOMAttributeNames: Record<string, string> = {
+  acceptCharset: 'accept-charset',
+  className: 'class',
+  htmlFor: 'for',
+  httpEquiv: 'http-equiv',
+  noModule: 'noModule',
+}
+
+const ignoreProps = [
+  'onLoad',
+  'onReady',
+  'dangerouslySetInnerHTML',
+  'children',
+  'onError',
+  'strategy',
+  'stylesheets',
+]
+
+function isBooleanScriptAttribute(
+  attr: string
+): attr is 'async' | 'defer' | 'noModule' {
+  return ['async', 'defer', 'noModule'].includes(attr)
+}
+
+export function setAttributesFromProps(el: HTMLElement, props: object) {
+  for (const [p, value] of Object.entries(props)) {
+    if (!props.hasOwnProperty(p)) continue
+    if (ignoreProps.includes(p)) continue
+
+    // we don't render undefined props to the DOM
+    if (value === undefined) {
+      continue
+    }
+
+    const attr = DOMAttributeNames[p] || p.toLowerCase()
+
+    if (el.tagName === 'SCRIPT' && isBooleanScriptAttribute(attr)) {
```

Comment:
```suggestion
    if (el.tagName === 'SCRIPT' && isBooleanScriptAttribute(attr) && value !== 'false') {
```
Couldn't we avoid setting the initial value here for the `'false'` case and then that should remove the need for the `setAttribute(attr, ''); removeAttribute(attr)` condition below?

Reply:
Unfortunately, this will have the same problem that this PR is trying to fix:
- `false`, this will enter this if condition and set the attribute to `false`, which HTML casts to `"false"` which is interpreted as true (non-empty string)
- `"false"`, this will enter the else condition and set the attribute to `"false"` which is interpreted as true (non-empty string)

This also breaks for script elements, which when inserted by JavaScript automatically default to async=true. So for async=false we need to do this setAttribute/removeAttribute trick explicitly. See https://html.spec.whatwg.org/multipage/scripting.html#script-force-async